dunamai/__init__.py

```diff
@@ -990,29 +991,62 @@ def get_version(
     :param third_choice: Callback to determine a version if the installed
         package cannot be found by name.
     :param fallback: If no other matches found, use this version.
+    :param ignore: Ignore this version if it is found.
     """
+    if isinstance(ignore, str):
+        ignore = Version(ignore)
+
     if first_choice:
         first_ver = first_choice()
-        if first_ver:
+        if first_ver and first_ver != ignore:
             return first_ver
 
     try:
         import importlib.metadata as ilm
     except ImportError:
         import importlib_metadata as ilm  # type: ignore
     try:
-        return Version(ilm.version(name))
+        ilm_version = Version(ilm.version(name))
+        if ilm_version != ignore:
+            return ilm_version
     except ilm.PackageNotFoundError:
         pass
 
     if third_choice:
         third_ver = third_choice()
-        if third_ver:
+        if third_ver and third_ver != ignore:
             return third_ver
 
     return fallback
 
 
+def get_version_from_vcs(
```

Comment:
I'd prefer not to add this.

* Writing `get_version("foo", third_choice=Version.from_any_vcs)` is simple enough if you need it and less commonly needed than using `Version.from_any_vcs` on its own, so I don't think it warrants adding a shorthand to the API surface.
* It's not clear from the name if the VCS would be the first or third choice, which could be confusing for users.

Reply:
I've removed the function